Resources/config/route_annotation_listener.xml

```diff
@@ -11,6 +11,7 @@
         <service id="presta_sitemap.eventlistener.route_annotation" class="%presta_sitemap.eventlistener.route_annotation.class%">
             <argument type="service" id="router"/>
             <argument>%presta_sitemap.default_section%</argument>
+            <argument type="collection">%presta_sitemap.alternate%</argument>
```

Comment:
Why `type=collection` ? Shouldn't it be simply `<argument>%presta_sitemap.alternate%</argument>` ?

Reply:
Because is an array of elements like this:

```yaml
 presta_sitemap:
     alternate:
         default_locale: 'en'
         locales: ['en', 'it']
         i18n: jms
 ```

I'm not sure is the same use `<argument>`

Reply:
@l-vo suggestion would work, but I prefer to see in the service declaration that this argument is an array, so it's ok for me

Reply:
Hum, does it work like that on your environments ? To me use `type="collection"` results to parse child nodes of ` <argument type="collection">` (see  [XmlFileLoader](https://github.com/symfony/symfony/blob/33e2735d1409554a040b3a293ee8e121c84248cb/src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php#L500)). But in this case there is no child node. So it would lead to receive an empty array as alternate argument; but maybe I miss something ?

Reply:
Using `<argument type="collection">` is only relevant when you are planning to define the collection structure manually : 

```
<argument type="collection">
    <argument key="foo">FOO</argument>
    <argument key="bar">BAR</argument>
</argument>
```

In our case, the value is injected from a parameter, there is no need for the `type="collection"`, I still encourage to do it, for the sake of maintenability.

Reply:
In this case, to me, it's not a matter of tase, this just doesn't work. Sorry for the noise if I'm wrong 🙂

Reply:
You tested it and it fails, or your did not tested it and you are expecting it to fail ?
Can't find any example in my memory, but I feel like I did it many times without any problem...

Reply:
I tested id but maybe I made a configuration mistake; it's why I didn't affirm I tested it. It's not a problem if the argument is empty (like `<argument type="collection" />`) and you replace it in a compiler pass, but in that case the parameter seems not present when the service configuration xml is loaded.

Reply:
If the parameter is not defined, no matter the structure of XML service configuration, the compilation will fail, because of that missing parameter